src/components/tl/ucc_tl.h

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "config.h"
+
+#ifndef UCC_TL_H_
+#define UCC_TL_H_
+
+#include "components/base/ucc_base_iface.h"
+
+/** TL (transport layer) interface abstracts the implementation
+  * of collective primitives over various tranports, such as:
+  * UCP (p2p based implementation), Sharp, Mcast, Shm, etc.
+  * TL interface is internal to UCC library and is used directly
+  * by different CL components.
+  * Different CLs may share TL context: e.g., CL_BASIC and CL_HIER
+  * will both use TL_UCP and share its context.
+  * A TL component may not use another TL (i.e. horizontal
+  * dependency is not allowed).
+  * The selection of TLs in runtime is defined by the usage at CL
+  * level.
+  */
+
```

Comment:
TL (transport layer) is an internal interface that provides a basic implementation of collectives and p2p primitives. It differs from CL in that it focuses on the abstraction over hardware transport rather than the programming model. These collectives can leverage either p2p communication transport or collective transport. The primary example of p2p communication transport is UCX, and the primary example of collective transport is SHARP, MCAST, and shared memory.

Reply:
applied